### PR TITLE
[selinux] Update checkpolicy libselinux, libsepol to v3.6

### PR DESCRIPTION
**Pull request by Camelron**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Update checkpolicy libselinux, libsepol to v3.6

###### Change Log  

- Update checkpolicy libselinux, libsepol to v3.6

###### Does this affect the toolchain?  

YES

###### Test Methodology

- buddy build succeeded

### CODE REVIEW

File: SPECS/checkpolicy/checkpolicy.spec [hunk 1]
```diff
@@ -1,8 +1,8 @@
-%define libselinuxver 3.2-1
-%define libsepolver 3.2-1
+%define libselinuxver 3.6-1
```
Comment (mfrw) [inline SPECS/checkpolicy/checkpolicy.spec:1]: I would suggest to link the buddy build link :)

File: SPECS/libsepol/libsepol.spec [hunk 2]
```diff
@@ -66,15 +66,6 @@ rm -f %{buildroot}%{_bindir}/chkcon
 rm -rf %{buildroot}%{_mandir}/man8
 rm -rf %{buildroot}%{_mandir}/ru/man8
 
-%check
```
Comment (cwize1) [inline SPECS/libsepol/libsepol.spec:69]: Why were the tests removed?

Comment (Camelron) [reply SPECS/libsepol/libsepol.spec:69]: The logic to unpack the "checkpolicy" project and run the tests does not make sense for libsepol. Instead, I just added the proper check section to our checkpolicy package which actually has the sources by default. 

See checkpolicy.spec change: https://github.com/microsoft/azurelinux/pull/7733/files#diff-329a0da3241ec2721fe29b0f1c416d30bb4d6191c292a1a514ccd5368ff6ac69R50

Comment (Camelron) [reply SPECS/libsepol/libsepol.spec:69]: So the same checks are being run, just in checkpolicy.

**Comment by Camelron** [comment]
Despite flaws in the buddybuild later on, the package builds succeed: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=516364&view=logs&j=7909073b-e471-53c1-2c44-627e2efc0564&t=fbeef549-39ab-510c-faa3-78579a516b53&l=1888

**Review by dmcilvaney — APPROVED** [review]

**Pull request merged**